sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateUnsafeProjection.scala

```diff
@@ -176,6 +141,23 @@ object GenerateUnsafeProjection extends CodeGenerator[Seq[Expression], UnsafePro
     """.trim
   }
 
+
+  private def writeArrayToBuffer(
```

Comment:
It's weird to have 2 `writeArrayToBuffer`. Since this one is only called in one place, can we inline it?